### PR TITLE
verify docker daemon is running in preflight checks

### PR DESCRIPTION
**Pull request by blaisdellk**
### Description 

https://github.com/confluentinc/cp-demo/issues/371

Verify docker daemon is running.

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

[ x] Run cp-demo (MacOS Catalina only)

### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._

 test on linux

### CODE REVIEW

File: scripts/helper/functions.sh [hunk 1]
```diff
@@ -41,6 +41,9 @@ preflight_checks()
     verify_installed $cmd || exit 1
   done
 
+  # Verify Docker daemon is running
+  docker ps -q || exit 1
```
Comment (ybyzek) [inline scripts/helper/functions.sh:45]: @blaisdellk if it's just an exit, user may not know why. Please add an error message (see above/below) for examples.  Perhaps something like `ERROR: cp-demo requires Docker to be running.  Please start Docker and run again.`

Comment (ybyzek) [reply scripts/helper/functions.sh:45]: @blaisdellk makes sense.

**Review by ybyzek — COMMENTED** [review]
@blaisdellk thanks for the PR, LGTM.  Before approving, I just wanted to highlight that the base branch being merged into is `master`, but most users are running latest (`6.2.0-post` at the moment).  As a result, users won't see this change until a future release.  Unless: consider applying this PR to `6.2.0-post` instead.

**Comment by blaisdellk** [comment]
Thanks @ybyzek.  I'm not aware of a way to alter the GitHub PR target so I created a https://github.com/confluentinc/cp-demo/pull/374 to supersede this PR.

**Pull request closed without merging**